base/threadingconstructs.jl

```diff
@@ -175,9 +175,56 @@ end
 function _threadsfor(iter, lbody, schedule)
     lidx = iter.args[1]         # index
     range = iter.args[2]
+    esc_range = esc(range)
+    func = if schedule === :greedy
+        greedy_func(esc_range, lidx, lbody)
+    else
+        default_func(esc_range, lidx, lbody)
+    end
     quote
         local threadsfor_fun
-        let range = $(esc(range))
+        $func
+        if $(schedule === :greedy || schedule === :dynamic || schedule === :default)
+            threading_run(threadsfor_fun, false)
+        elseif ccall(:jl_in_threaded_region, Cint, ()) != 0 # :static
+            error("`@threads :static` cannot be used concurrently or nested")
+        else # :static
+            threading_run(threadsfor_fun, true)
+        end
+        nothing
+    end
+end
+
+function greedy_func(itr, lidx, lbody)
+    quote
+        let r = Ref{Task}(), c = Channel{eltype($itr)}(threadpoolsize(),taskref=r) do ch
```

Comment:
```suggestion
        let r = Ref{Task}(), c = Channel{eltype($itr)}(0,taskref=r) do ch
```
This is not exactly greedy unless the lazy buffer is disabled

Reply:
```suggestion
        let r = Ref{Task}(), c = Channel{eltype($itr)}(threadpoolsize(),taskref=r,spawn=false) do ch
```
It is also not threaded unless directed to be

Reply:
`spawn=true` is already the default here, no?

Reply:
It's greedy in the sense that the consumer tasks greedily take the next work item.